docs/architecture/overview.md

```diff
@@ -0,0 +1,193 @@
+
+DCP is written in Go, aligning with Kubernetes and its ecosystem, which are also Go-based. This choice enables deep, native integration with Kubernetes APIs, efficient concurrency, and access to mature tooling like Kubebuilder. DCP is delivered as two executables:
+
+- `dcp.exe`: API server that exposes a Kubernetes-like API endpoint for the app host to communicate with. Additionally, it exposes log streaming to the app host, which ultimately streams logs to the developer dashboard.
+- `dcpctrl.exe`: Controller that monitors the API server for new objects and changes, ensuring that the real-world environment matches the specified model.
+
+> [!NOTE]
+> DCP strives for "eventual consistency" rather than "strong consistency." This means that while DCP aims to keep the real-world environment in sync with the specified model, it might not always be perfectly aligned.
+
+When you run the app host, it communicates with DCP using Kubernetes client libraries. The app host gives the app model to DCP, which looks at the resources in the model and turns them into specs it can understand. This process involves translating the app model into a set of Kubernetes custom resource definitions (CRDs) that represent the desired state of the application
+
+DCP performs the following tasks:
+
+- Prepares the resources for execution:
+  - Configures service endpoints.
+    - Assigns names and ports dynamically, unless explicitly set (DCP ensures that the ports are available and not in use by other processes).
+  - Initializes container networks.
+  - Pulls container images based on their applied [pull policy](xref:Aspire.Hosting.ApplicationModel.ImagePullPolicy).
+  - Creates and starts containers.
+  - Runs executables with the required arguments and environment variables.
+- Monitors resources:
+  - Provides change notifications about objects managed within DCP, including process IDs, running status, and exit codes (the app host subscribes to these changes to manage the [application's lifecycle](../app-host/eventing.md) effectively).
+- Starts the developer dashboard.
+
+Continuing from the [diagram in the previous](#app-host-dcp-flow) section, consider the following diagram that helps to visualize the responsibilities of DCP:
+
+<span id="dcp-architecture"></span>
+
+:::image type="content" source="media/dcp-architecture-thumb.png" alt-text="A diagram depicting the architecture of the Developer Control Plane (DCP)." lightbox="media/dcp-architecture.png":::
+
+DCP logs are streamed back to the app host, and then the app host pushes them to the developer dashboard. DCP registers default commands for the developer dashboard as well, such as start, stop, and restart. For more information on commands, see [Custom resource commands in .NET Aspire](../fundamentals/custom-resource-commands.md).
```

Comment:
There is no "command" concept in DCP. Dashboard commands are implemented by the app model runtime, the "dashboard service" part of it specifically. Under the covers, these commands manipulate DCP objects (create new ones, delete old ones, and set properties). For example, a .NET project restart involves stopping and deleting the old Executable representing the project and creating a new one with identical Spec.